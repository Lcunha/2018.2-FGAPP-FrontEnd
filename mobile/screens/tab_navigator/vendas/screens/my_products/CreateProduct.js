/*
    Screen provided to present the products already ordered by a certain user.
*/

import React, { Component } from 'react';
import {
    View,
    StyleSheet,
    Text
} from 'react-native';
import ProductImage from '../../components/ProductImage';
import { Container, Header, Textarea, Content, Alert, Form, Item, Input, Label, Button } from 'native-base';
import jwt_decode from 'jwt-decode';
import ErrorDialog from './ErrorDialog';

class CreateProduct extends Component {
    constructor(props) {
      super(props);
      this.state = {
        title: '',
        price: '',
        description: '',
        isDialogVisible:false,
        messageError: '',
      };
    }

    openDialog = async () => {
      this.setState({ isDialogVisible: true })
    }
    closeDialog = async () => {
      this.setState({ isDialogVisible: false })
    }

    registerProduct = async () => {
      const {state} = this.props.navigation;
      var token = state.params ? state.params.token : undefined;
      var jwtDecode = require('jwt-decode');
      var user = jwt_decode(token);
      fetch(`${process.env.VENDAS_API}/api/create_product/`, {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
        },
        body: JSON.stringify({
          'fk_vendor': user.user_id,
          'name': this.state.title,
          'price': this.state.price,
          'photo': 'https://cdn.cnn.com/cnnnext/dam/assets/171027052520-processed-foods-exlarge-tease.jpg',
          'description': this.state.description,
        }),
      })
      .then((response) => {
        return response.json();
      })
      .then((responseJson) => {
        if (responseJson.length){
          const error = JSON.parse(responseJson[0]);
          this.setState ({ messageError: error["error"]})
          this.setState({ isDialogVisible: true })
        }
        else{
          this.props.navigation.navigate('MyProducts', {token:token});
        }
      })
      .catch((err) => {
        this.setState ({ messageError: "Erro interno, não foi possível se comunicar com o servidor."})
        this.setState({ isDialogVisible: true })
      })
    }


    render() {
        const {state} = this.props.navigation;
        var token = state.params ? state.params.token : undefined;
        var jwtDecode = require('jwt-decode');
        var user = jwt_decode(token);
        return (
            <View style={styles.container}>
              <ErrorDialog
                  messageError={this.state.messageError}
                  isDialogVisible={this.state.isDialogVisible}
                  backButton = {this.closeDialog}
              />
              <ProductImage style= { styles.image } photo='http://www.piniswiss.com/wp-content/uploads/2013/05/image-not-found-4a963b95bf081c3ea02923dceaeb3f8085e1a654fc54840aac61a57a60903fef-300x199.png'>
              </ProductImage>
              <Form style={styles.description}>
                <Item floatingLabel>
                  <Label>Título</Label>
                  <Input style={{ color: "white" }} onChangeText={(title) => {this.setState({title})}} />
                </Item>
                <Item floatingLabel>
                  <Label>Preço</Label>
                  <Input style={{ color: "white" }} onChangeText={(price) => {this.setState({price})}} />
                </Item>
                <Textarea style={{ color: "white" }} onChangeText={(description) => {this.setState({description})}} rowSpan={2} underline placeholder="Descrição" />
              </Form>
              <View style={styles.buttonContainer}>
                <Button onPress={() => {this.props.navigation.navigate('MyProducts', {token:token});}} style={{justifyContent: 'center', height: 40, width: 100}} danger><Text style={{color: 'white'}}> CANCELAR </Text></Button>
                <Button onPress={this.registerProduct} style={{justifyContent: 'center', height: 40, width: 100}} success><Text style={{color: 'white'}}> SALVAR </Text></Button>
              </View>

            </View>
        );
    }
}
export default CreateProduct;

const styles = StyleSheet.create({
    container: {
      backgroundColor: '#171717',
        flex: 1
    },
    image: {
      width: '100%',
      height: '30%',
    },
    description: {
      flex: 1,
      height: '35%',
      width: '95%',
    },
    buttonContainer: {
      flexDirection: 'row',
      justifyContent: 'space-around',
      paddingBottom: 10,
    },
});
