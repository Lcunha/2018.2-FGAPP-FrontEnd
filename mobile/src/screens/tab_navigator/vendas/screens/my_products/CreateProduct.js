/*
    Screen provided for a seller create a product.
*/

import React, { Component } from 'react';
import {
    View,
    StyleSheet,
    Text,
    Keyboard,
    Animated
} from 'react-native';
import ProductImage from '../../components/ProductImage';
import { Textarea, Form, Item, Input, Label, Button } from 'native-base';
import jwt_decode from 'jwt-decode';
import ErrorDialog from './ErrorDialog';
import ToogleView from './ToogleView';
import {getUserToken} from '../../../../../AuthMethods'

class CreateProduct extends Component {
    constructor(props) {
      super(props);
      this.keyboardHeight = new Animated.Value(0);
      this.imageHeight = new Animated.Value(199);
      this.state = {
        token:undefined,
        isButtonsHidden: false,
        title: '',
        price: '',
        description: '',
        isDialogVisible: false,
        messageError: '',
      };
    }
    componentWillMount(){
      getUserToken()
      .then(res => this.setState({ token: res }))
      .catch(err => alert("Erro"));
    }

    openDialog = async () => {
      this.setState({ isDialogVisible: true })
    }
    closeDialog = async () => {
      this.setState({ isDialogVisible: false })
    }

    registerProduct = async () => {
      var user = jwt_decode(this.state.token);
      const create_product_path = `${process.env.VENDAS_API}/api/create_product/`;
      fetch(create_product_path, {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
        },
        body: JSON.stringify({
          'fk_vendor': user.user_id,
          'name': this.state.title,
          'price': this.state.price,
          'photo': 'https://foodrevolution.org/wp-content/uploads/2018/04/blog-featured-diabetes-20180406-1330.jpg',
          'description': this.state.description,
          'token':this.state.token,
        }),
      })
      .then((response) => {
        return response.json();
      })
      .then((responseJson) => {
        if (responseJson.error != undefined){
          this.setState ({ messageError: responseJson.error})
          this.setState({ isDialogVisible: true })
        }
        else{
          this.props.navigation.navigate('MyProducts');
        }
      })
      .catch((err) => {
        this.setState ({ messageError: "Erro interno, não foi possível se comunicar com o servidor."})
        this.setState({ isDialogVisible: true })
      })
    }

    componentDidMount () {
      this.keyboardDidShowListener = Keyboard.addListener('keyboardDidShow', this._keyboardDidShow);
      this.keyboardDidHideListener = Keyboard.addListener('keyboardDidHide', this._keyboardDidHide);
    }

    componentWillUnmount () {
      this.keyboardDidShowListener.remove();
      this.keyboardDidHideListener.remove();
    }

    _keyboardDidShow = (event) => {
      this.setState({ isButtonsHidden: true });
      Animated.parallel([
        Animated.timing(this.keyboardHeight, {
          duration: event.duration,
          toValue: event.endCoordinates.height,
        }),
        Animated.timing(this.imageHeight, {
          duration: event.duration,
          toValue: 0,
        }),
      ]).start();
    }

    _keyboardDidHide = (event) => {
      this.setState({ isButtonsHidden: false });
      Animated.parallel([
        Animated.timing(this.keyboardHeight, {
          toValue: 0,
        }),
        Animated.timing(this.imageHeight, {
          toValue: 199,
        }),
      ]).start();
    }

    render() {

        return (
            <Animated.View style={[styles.container, { paddingBottom: this.keyboardHeight }]}>
              <ErrorDialog
                  messageError={this.state.messageError}
                  isDialogVisible={this.state.isDialogVisible}
                  backButton = {this.closeDialog}
              />
              <Animated.Image source={{ uri: 'http://www.piniswiss.com/wp-content/uploads/2013/05/image-not-found-4a963b95bf081c3ea02923dceaeb3f8085e1a654fc54840aac61a57a60903fef-300x199.png' }} style={[styles.logo, { height: this.imageHeight, width: '100%' }]} />
              <Form style={styles.description}>
                <Item floatingLabel>
                  <Label>Título</Label>
                  <Input
                    style={{ color: 'black' }}
                    onChangeText={(title) => {this.setState({title})}}
                  />
                </Item>
                <Item floatingLabel>
                  <Label>Preço</Label>
                  <Input
                    style={{ color: 'black' }}
                    keyboardType='numeric'
                    onChangeText={(price) => {this.setState({price})}}
                  />
                </Item>
                <Textarea
                  style={{ color: 'black' }}
                  onChangeText={(description) => {this.setState({description})}}
                  rowSpan={2}
                  underline
                  placeholder="Descrição"
                />
              </Form>
              <ToogleView hide={this.state.isButtonsHidden}>
                <View style={styles.buttonContainer}>
                  <Button
                    onPress={() => {this.props.navigation.navigate('MyProducts');}}
                    style={styles.button}
                    danger
                  >
                    <Text style={{color: 'white'}}> CANCELAR </Text>
                  </Button>
                  <Button
                    onPress={this.registerProduct}
                    style={styles.button}
                    success
                  >
                    <Text style={{color: 'white'}}> SALVAR </Text>
                  </Button>
                </View>
              </ToogleView>
            </Animated.View>
        );
    }
}
export default CreateProduct;

const styles = StyleSheet.create({
    container: {
      backgroundColor: 'white',
        flex: 1
    },
    description: {
      flex: 1,
      height: '35%',
      width: '95%',
    },
    buttonContainer: {
      flexDirection: 'row',
      justifyContent: 'space-around',
      paddingBottom: 10,
    },
    button: {
      justifyContent: 'center',
      height: 40,
      width: 100,
    }
});
