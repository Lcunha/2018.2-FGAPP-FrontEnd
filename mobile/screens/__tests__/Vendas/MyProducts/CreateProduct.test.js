import React from 'react';
import CreateProduct from '../../../tab_navigator/vendas/screens/my_products/CreateProduct';
import Adapter from 'enzyme-adapter-react-16';
import {shallow} from 'enzyme';
import Enzyme from 'enzyme';
import renderer from 'react-test-renderer';
import fetchMock from 'fetch-mock';

Enzyme.configure({adapter: new Adapter()});

function FormDataMock(){
  this.append = jest.fn();
}
global.FormData = FormDataMock;

describe('Testing navigation', () => {

  let wrapper = null
  const spyNavigate = jest.fn()
  const props = {
    navigation: {
      navigate: spyNavigate,
      state: {}
    }
  }
  const params = {
    token: 'randomToken'
  }

  beforeEach(() => {
    wrapper = shallow(<CreateProduct {...props} />)
    wrapper.setState({ params: params })
  })

  it('should test navigation', async () => {
    await wrapper.instance()._goBack(params)
    expect(spyNavigate).toHaveBeenCalled()
  })
})

describe('Test CreateProduct', () => {
    const navigation = {
        state: {
            params: {
                token: "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJlbWFpbCI6InJvZ2VybGVua2VAZ21haWwuY29tIiwidXNlcl9pZCI6MSwib3JpZ19pYXQiOjE1NDE3MTk3NDksImV4cCI6MTU0MTcyMDA0OSwidXNlcm5hbWUiOiJyb2dlcmxlbmtlQGdtYWlsLmNvbSJ9.eCEGRB9yYAkP5iBIybeDsAoWk4HyusPUTX3LBiP0I64"
            }
        },
        navigate: jest.fn(),
    }

    // Url to be mocked
    let create_product_path = '';

    beforeAll(() => {
        process.env.VENDAS_API = 'http://test.ip';
        create_product_path = `${process.env.VENDAS_API}/api/create_product/`;
    })

    beforeEach(() => {
        fetchMock.restore();
    })

    it('renders correctly', () => {
      const navigation = jest.fn();
      const tree = renderer.create(<CreateProduct navigation={navigation}/>).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('Test registerProduct with sucess', async (done) => {
        const wrapper = shallow(<CreateProduct navigation={navigation}/>);

        const state = {
            title: "Delicious carrot cake",
            price: 1,
            description: "Not only delicious carrot cake, but with chocolate cover"
        };

        fetchMock.post(create_product_path, []);

        wrapper.setState(state);
        await wrapper.instance().registerProduct()

        process.nextTick(() => {
            done();
        });
    });

    it('Test registerProduct with error response', async (done) => {
        const wrapper = shallow(<CreateProduct navigation={navigation}/>);

        const state = {
            title: "Delicious carrot cake",
            price: 1,
            description: "Not only delicious carrot cake, but with chocolate cover"
        };

        const error = "Campo inválido de preço";

        fetchMock.post(create_product_path, { error });

        wrapper.setState(state);
        await wrapper.instance().registerProduct()

        process.nextTick(() => {
            wrapper.update();
            done();
        });
    })

    it('Test registerProduct with error', async (done) => {
        const wrapper = shallow(<CreateProduct navigation={navigation}/>);

        const state = {
            title: "Delicious cake",
            price: 1,
            description: "There is no chocolate cover, but it still delicious"
        };

        fetchMock.post(create_product_path, 500);

        wrapper.setState(state);
        await wrapper.instance().registerProduct();

        process.nextTick(() => {
            wrapper.update();
            done();
        });
    });

    it('tests openDialog to be true',() => {
      const navigation = jest.fn();
      const wrapper = shallow(<CreateProduct navigation = {navigation}/>);
      const openDialog = wrapper.instance().openDialog();
    });

    it('tests closeDialog to be false',() => {
      const navigation = jest.fn();
      const wrapper = shallow(<CreateProduct navigation = {navigation}/>);
      const closeDialog = wrapper.instance().closeDialog();
    });

    it('tests goBack function',() => {
      const navigation = jest.fn();
      const wrapper = shallow(<CreateProduct navigation = {navigation}/>);
      const _goBack = wrapper.instance()._goBack();
    });

    it('tests componentWillUnmount function',() => {
      const navigation = jest.fn();
      const wrapper = shallow(<CreateProduct navigation = {navigation}/>);
      const componentWillUnmount = wrapper.instance().componentWillUnmount();
    });

    it('test keyboardDidHide function',() => {
      const navigation = jest.fn();
      const wrapper = shallow(<CreateProduct navigation = {navigation}/>);
      const _keyboardDidHide = wrapper.instance()._keyboardDidHide();
    });

    test('change input title ', () => {
      const wrapper = shallow(<CreateProduct />);
      const title = wrapper.find('InputLab').at(0);
      title.simulate('changeText','text');
      expect(wrapper.state('title')).toBe('text');
    });

    test('change input price ', () => {
      const wrapper = shallow(<CreateProduct />);
      const price = wrapper.find('InputLab').at(1);
      price.simulate('changeText','text');
      expect(wrapper.state('price')).toBe('text');
    });

    test('change text area description ', () => {
      const wrapper = shallow(<CreateProduct />);
      const description = wrapper.find('Styled(Textarea)').at(0);
      description.simulate('changeText','text');
      expect(wrapper.state('description')).toBe('text');
    });
})
