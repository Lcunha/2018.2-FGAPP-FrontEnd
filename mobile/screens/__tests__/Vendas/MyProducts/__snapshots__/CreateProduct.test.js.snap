// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test CreateProduct renders correctly 1`] = `
<View
  collapsable={undefined}
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "paddingBottom": 0,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "column",
      }
    }
  >
    <Modal
      animationType="none"
      hardwareAccelerated={false}
      hideModalContentWhileAnimating={false}
      onModalHide={[Function]}
      onRequestClose={[Function]}
      scrollOffset={0}
      scrollOffsetMax={0}
      scrollTo={null}
      supportedOrientations={
        Array [
          "portrait",
          "landscape",
        ]
      }
      swipeThreshold={100}
      transparent={true}
      visible={false}
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "black",
            "bottom": 0,
            "height": 1334,
            "left": 0,
            "opacity": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 750,
          }
        }
        testID={undefined}
      />
      <View
        collapsable={undefined}
        hideModalContentWhileAnimating={false}
        onModalHide={[Function]}
        pointerEvents="box-none"
        scrollOffset={0}
        scrollOffsetMax={0}
        scrollTo={null}
        style={
          Object {
            "flex": 1,
            "justifyContent": "center",
            "margin": 37.5,
            "marginBottom": 0,
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": 0,
            "transform": Array [
              Object {
                "translateY": 0,
              },
            ],
          }
        }
        supportedOrientations={
          Array [
            "portrait",
            "landscape",
          ]
        }
        swipeThreshold={100}
      >
        <View
          enabled={true}
          keyboardVerticalOffset={0}
          onLayout={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              },
              Object {
                "paddingBottom": 0,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "borderRadius": 13,
                "flexDirection": "column",
                "overflow": "hidden",
                "width": 270,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "rgba(255,255,255, 0.8)",
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            />
            <View
              style={
                Object {
                  "margin": 18,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "black",
                      "fontSize": 18,
                      "fontWeight": "600",
                      "textAlign": "center",
                    },
                    Object {
                      "paddingBottom": 5,
                      "paddingTop": 0,
                    },
                  ]
                }
              >
                Erro
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "black",
                      "fontSize": 13,
                      "marginTop": 4,
                      "textAlign": "center",
                    },
                    undefined,
                  ]
                }
              >
                
              </Text>
            </View>
            <View
              style={
                Object {
                  "borderTopColor": "#A9ADAE",
                  "borderTopWidth": 0.5,
                  "flexDirection": "row",
                  "height": 46,
                  "justifyContent": "space-between",
                }
              }
            >
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                collapsable={undefined}
                hasTVPreferredFocus={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "flex": 1,
                    "justifyContent": "center",
                    "opacity": 1,
                  }
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "fontSize": 17,
                        "textAlign": "center",
                      },
                      Object {
                        "color": "purple",
                        "fontWeight": "normal",
                      },
                      undefined,
                    ]
                  }
                >
                  Cancelar
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </Modal>
  </View>
  <Image
    collapsable={undefined}
    source={
      Object {
        "uri": "http://www.piniswiss.com/wp-content/uploads/2013/05/image-not-found-4a963b95bf081c3ea02923dceaeb3f8085e1a654fc54840aac61a57a60903fef-300x199.png",
      }
    }
    style={
      Object {
        "height": 199,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Array [
        Object {},
        Object {
          "flex": 1,
          "height": "35%",
          "width": "95%",
        },
      ]
    }
    virtual={undefined}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderBottomWidth": 1,
          "borderColor": "#D9D5DC",
          "borderLeftWidth": 0,
          "borderRadius": undefined,
          "borderRightWidth": 0,
          "borderTopWidth": 0,
          "borderWidth": undefined,
          "flexDirection": "row",
          "marginLeft": 15,
          "marginTop": 15,
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        collapsable={undefined}
        style={
          Object {
            "left": 0,
            "opacity": 1,
            "paddingBottom": undefined,
            "paddingTop": undefined,
            "position": "absolute",
            "right": 0,
            "top": 18,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#575757",
              "fontSize": 17,
              "left": 0,
              "paddingRight": 5,
              "top": 6,
            }
          }
          virtual={undefined}
        >
          Título
        </Text>
      </View>
      <TextInput
        allowFontScaling={true}
        editable={true}
        nameLabel="Título"
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholderTextColor="#575757"
        style={
          Array [
            Object {
              "color": "#000",
              "flex": 1,
              "fontSize": 17,
              "height": 50,
              "marginLeft": null,
              "paddingBottom": 7,
              "paddingLeft": null,
              "paddingRight": 5,
              "paddingTop": 3,
              "top": 10,
            },
            Object {
              "color": "black",
            },
          ]
        }
        underlineColorAndroid="rgba(0,0,0,0)"
        value=""
        virtual={undefined}
      />
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderBottomWidth": 1,
          "borderColor": "#D9D5DC",
          "borderLeftWidth": 0,
          "borderRadius": undefined,
          "borderRightWidth": 0,
          "borderTopWidth": 0,
          "borderWidth": undefined,
          "flexDirection": "row",
          "marginLeft": 15,
          "marginTop": 15,
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        collapsable={undefined}
        style={
          Object {
            "left": 0,
            "opacity": 1,
            "paddingBottom": undefined,
            "paddingTop": undefined,
            "position": "absolute",
            "right": 0,
            "top": 18,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#575757",
              "fontSize": 17,
              "left": 0,
              "paddingRight": 5,
              "top": 6,
            }
          }
          virtual={undefined}
        >
          Preço
        </Text>
      </View>
      <TextInput
        allowFontScaling={true}
        editable={true}
        keyboardType="numeric"
        nameLabel="Preço"
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholderTextColor="#575757"
        style={
          Array [
            Object {
              "color": "#000",
              "flex": 1,
              "fontSize": 17,
              "height": 50,
              "marginLeft": null,
              "paddingBottom": 7,
              "paddingLeft": null,
              "paddingRight": 5,
              "paddingTop": 3,
              "top": 10,
            },
            Object {
              "color": "black",
            },
          ]
        }
        underlineColorAndroid="rgba(0,0,0,0)"
        value=""
        virtual={undefined}
      />
    </View>
    <TextInput
      allowFontScaling={true}
      editable={true}
      multiline={true}
      onChangeText={[Function]}
      placeholder="Descrição"
      placeholderTextColor="#575757"
      rowSpan={2}
      style={
        Object {
          "borderBottomWidth": 0.5,
          "borderColor": "#D9D5DC",
          "color": "black",
          "fontSize": 15,
          "height": 50,
          "marginTop": 5,
          "paddingLeft": 10,
          "paddingRight": 5,
          "textAlignVertical": "top",
        }
      }
      underline={true}
      underlineColorAndroid="rgba(0,0,0,0)"
      virtual={undefined}
    />
  </View>
  <View
    style={undefined}
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-around",
          "paddingBottom": 10,
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "#830037",
            "flex": 0,
            "height": 50,
            "justifyContent": "center",
            "opacity": 1,
            "width": 100,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "alignItems": "center",
              "color": "white",
              "fontSize": 16,
              "fontWeight": "300",
            }
          }
        >
           
          CANCELAR
           
        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "#0EAC6F",
            "flex": 0,
            "height": 50,
            "justifyContent": "center",
            "opacity": 1,
            "width": 100,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "alignItems": "center",
              "color": "white",
              "fontSize": 16,
              "fontWeight": "300",
            }
          }
        >
           
          SALVAR
           
        </Text>
      </View>
    </View>
  </View>
</View>
`;
